File: typescript/src/renderer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.newRenderer = void 0;
function newRenderer(rectangle) {
    return {
        draw: function () {
            var area = rectangle.computeArea();
            console.log("Shape drawn\n" + "Its area is " + area);
        }
    };
}
exports.newRenderer = newRenderer;

File: java/src/main/java/edu/cmu/cs/cs214/lab02/Renderer.java
package edu.cmu.cs.cs214.lab02;

import edu.cmu.cs.cs214.lab02.shapes.*;

public class Renderer {
    private Rectangle rectangle;
    private double height;
    private double width;

    
    Renderer(double height, double width) {
        this.height = height;
        this.width = width;
        this.rectangle = new Rectangle(height, width);
    }

    void draw() {
        double area = rectangle.getArea();

        // assume implementation

        System.out.println("Shape printed\n" + "Its area is " + area);
    }
}
